### PR TITLE
Add support for MDX

### PR DESCRIPTION
**Pull request by asturur**
MDX allows for react component in MD.
That means we can keep the simplicity of MD for text and paragraphs, while adding example snippets with react.

All current demos do not work, and  also something strange is going on with the menu.

Need more work but is not entirely broken

### CODE REVIEW

**Comment by asturur** [comment]
todo: fix the code snippets in mdx not visibile

**Comment by asturur** [comment]
original: 
image

now:
image

File: gatsby-config.js [hunk 1]
```diff
@@ -51,14 +51,14 @@ module.exports = {
...
-            options: {
-              inlineCodeMarker: '>',
-            },
-          },
+          // {
+          //   resolve: 'gatsby-remark-prismjs',
+          //   options: {
+          //     inlineCodeMarker: '>',
+          //   },
+          // },
```
Comment (asturur) [inline gatsby-config.js:59]: need to fix this

**Comment by ShaMan123** [comment]
I pushed a commit for auto refresh on the canvas - the only thing left is to handle errors somehow, I tried trycatch block but that didn't help for some reason

**Comment by asturur** [comment]
should be an option with a checkbox maybe. The runme button should stay anyway, it tells the user he can edit the code

**Comment by asturur** [comment]
merging this, your fixes, then progressing with more onversions.

**Comment by ShaMan123** [comment]
remind me why https://www.gatsbyjs.com/plugins/gatsby-remark-codemirror/ isn't good enough please

**Comment by asturur** [comment]
i don't think we ever talked about it. i don't remember why. I have seen it but for
some reason i skipped it

**Comment by asturur** [comment]
oh is not an editor is just syntax highlighting

**Pull request merged**